ios/KaMPKitiOS/BreedListScreen.swift

```diff
@@ -9,77 +9,22 @@
 import Combine
 import SwiftUI
 import shared
+import KMMViewModelSwiftUI
 
 private let log = koin.loggerWithTag(tag: "ViewController")
 
-class ObservableBreedModel: ObservableObject {
-    private var viewModel: BreedCallbackViewModel?
-
-    @Published
-    var loading = false
-
-    @Published
-    var breeds: [Breed]?
-
-    @Published
-    var error: String?
-
-    private var cancellables = [AnyCancellable]()
-
-    func activate() {
-        let viewModel = KotlinDependencies.shared.getBreedViewModel()
-
-        doPublish(viewModel.breeds) { [weak self] dogsState in
-            self?.loading = dogsState.isLoading
-            self?.breeds = dogsState.breeds
-            self?.error = dogsState.error
-
-            if let breeds = dogsState.breeds {
-                log.d(message: {"View updating with \(breeds.count) breeds"})
-            }
-            if let errorMessage = dogsState.error {
-                log.e(message: {"Displaying error: \(errorMessage)"})
-            }
-        }.store(in: &cancellables)
-
-        self.viewModel = viewModel
-    }
-
-    func deactivate() {
-        cancellables.forEach { $0.cancel() }
-        cancellables.removeAll()
-
-        viewModel?.clear()
-        viewModel = nil
-    }
-
-    func onBreedFavorite(_ breed: Breed) {
-        viewModel?.updateBreedFavorite(breed: breed)
-    }
-
-    func refresh() {
-        viewModel?.refreshBreeds()
-    }
-}
-
 struct BreedListScreen: View {
-    @StateObject
-    var observableModel = ObservableBreedModel()
+    @StateViewModel
+    var observableModel = KotlinDependencies.shared.getBreedViewModel()
 
     var body: some View {
         BreedListContent(
-            loading: observableModel.loading,
-            breeds: observableModel.breeds,
-            error: observableModel.error,
-            onBreedFavorite: { observableModel.onBreedFavorite($0) },
-            refresh: { observableModel.refresh() }
+            loading: observableModel.breedState.isLoading,
+            breeds: observableModel.breedState.breeds,
+            error: observableModel.breedState.error,
+            onBreedFavorite: { observableModel.updateBreedFavorite(breed: $0) },
+            refresh: { observableModel.refreshBreeds() }
         )
-        .onAppear(perform: {
-            observableModel.activate()
-        })
-        .onDisappear(perform: {
-            observableModel.deactivate()
-        })
```

Comment:
Everything seems to be working as expected. Though I wasn't able to identify when `deactivate` was triggered (there is currently no alternative in KMM-ViewModel).